src/lazy/any_encoding.rs

```diff
@@ -181,10 +182,10 @@ impl<'data> From<RawStreamItem<'data, BinaryEncoding>> for RawStreamItem<'data,
 }
 
 impl<'data> LazyRawValuePrivate<'data> for LazyRawAnyValue<'data> {
-    fn field_name(&self) -> Option<RawSymbolTokenRef<'data>> {
+    fn field_name(&self) -> IonResult<RawSymbolTokenRef<'data>> {
```

Comment:
🗺️ This trait method signature had to be changed. The binary raw reader only ever encounters `SymbolId` field names. The text reader, however, can encounter string and symbol field names with invalid text. (For example: illegal escape sequences or invalid unicode.) Thus, we now return an `IonResult` instead of an `Option`.